plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ResourceContainer.java

```diff
@@ -0,0 +1,30 @@
+package org.jboss.tools.openshift.internal.ui.models;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.openshift.restclient.model.IResource;
+
+public abstract class ResourceContainer<R extends IResource, P extends IOpenshiftUIElement<?>>
+		extends AbstractResourceWrapper<R, P> {
+	protected Map<IResource, AbstractResourceWrapper<?, ?>> containedResources = new HashMap<>();
+	protected Object childrenLock = new Object();
+
+	public ResourceContainer(P parent, R resource) {
+		super(parent, resource);
+	}
+
+	public Collection<AbstractResourceWrapper<?, ?>> getResourcesOfKind(String kind) {

```

Comment:
I know this is not optimal. I posit it makes no perceivable difference to performance.